monai/metrics/active_learning_metrics.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+
+from monai.metrics.utils import ignore_background
+
+from .metric import Metric
+
+
+class VarianceMetric(Metric):
+    """
+    Compute the Variance of a given T-repeats N-dimensional array/tensor. The primary usage is as a uncertainty based
+    metric for Active Learning.
+
+    It can return the spatial variance/uncertainty map based on user choice or a single scalar value via mean/sum of the
+    variance for scoring purposes
+
+    Args:
+        include_background: Whether to include the background of the spatial image or channel 0 of the 1-D vector
+        spatial_map: Boolean, if set to True, spatial map of variance will be returned corresponding to i/p image dimensions
+        scalar_reduction: reduction type of the metric, either 'sum' or 'mean' can be used
+        threshold: To avoid NaN's a threshold is used to replace zero's
+
+    """
+
+    def __init__(
+        self,
+        include_background: bool = True,
+        spatial_map: bool = False,
+        scalar_reduction: str = "sum",
+        threshold: float = 0.0005,
+    ) -> None:
+        super().__init__()
+        self.include_background = include_background
+        self.spatial_map = spatial_map
+        self.scalar_reduction = scalar_reduction
+        self.threshold = threshold
+
+    def _compute(self, y_pred: torch.Tensor):
+        """
+        Args:
+            y_pred: Predicted segmentation, typically segmentation model output.
+                It must be N-repeats, repeat-first tensor [N,C,H,W,D].
+
+        Returns:
+            Pytorch tensor of scalar value of variance as uncertainty or a spatial map of uncertainty
+
+        """
+        return compute_variance(
+            y_pred=y_pred,
+            include_background=self.include_background,
+            spatial_map=self.spatial_map,
+            scalar_reduction=self.scalar_reduction,
+            threshold=self.threshold,
+        )
+
+
+def compute_variance(
+    y_pred: torch.Tensor, include_background=True, spatial_map=False, scalar_reduction="mean", threshold=0.0005
```

Comment:
Please add the typehints for these args.

Thanks.

Reply:
Thank you will add metric to doc and add typehints @Nic-Ma

Reply:
All added & addressed